app/assets/javascripts/components/utils.js.coffee

```diff
@@ -43,90 +43,65 @@ class @Utils
-        form.submit (e) =>
-          e.preventDefault()
-          json = $(e.target).find('.payload-editor').val()
-          json = '{}' if json == ''
-          try
-            payload = JSON.parse(json)
-            throw true unless payload.constructor is Object
-            if Object.keys(payload).length == 0
-              json = ''
-            else
-              json = JSON.stringify(payload)
-          catch
-            alert 'Invalid JSON object.'
-            return
-          if json == ''
-            if with_event_mode is 'yes'
-              alert 'Event is required for this agent to run.'
-              return
-            dry_run_data = data
-            $(button).data('payload', null)
-          else
-            dry_run_data = "event=#{encodeURIComponent(json)}&#{data}"
-            $(button).data('payload', json)
-          $(body).closest('[role=dialog]').on 'hidden.bs.modal', =>
-            @invokeDryRun(url, dry_run_data, cleanup)
-          .modal('hide')
-        $(body).closest('[role=dialog]').on 'shown.bs.modal', ->
-          $(this).find('.btn-primary').focus()
-      title: 'Dry Run'
-      onHide: cleanup
+            if previous = $(button).data('payload')
+              payload_editor.text(previous)
+
+            editor = window.setupJsonEditor(payload_editor)[0]
+
+            $(body).find('.dry-run-event-sample').click (e) =>
+              e.preventDefault()
+              editor.json = $(e.currentTarget).data('payload')
+              editor.rebuild()
+
+            form.submit (e) =>
+              e.preventDefault()
+              json = $(e.target).find('.payload-editor').val()
+              json = '{}' if json == ''
+              try
+                payload = JSON.parse(json)
+                throw true unless payload.constructor is Object
+                if Object.keys(payload).length == 0
+                  json = ''
+                else
+                  json = JSON.stringify(payload)
+              catch
+                alert 'Invalid JSON object.'
+                return
+              if json == ''
+                if with_event_mode is 'yes'
+                  alert 'Event is required for this agent to run.'
+                  return
+                dry_run_data = data
+                $(button).data('payload', null)
+              else
+                dry_run_data = "event=#{encodeURIComponent(json)}&#{data}"
+                $(button).data('payload', json)
+              $(body).closest('[role=dialog]').on 'hidden.bs.modal', =>
+                @invokeDryRun(url, dry_run_data, cleanup)
+              .modal('hide')
+            $(body).closest('[role=dialog]').on 'shown.bs.modal', ->
+              $(this).find('.btn-primary').focus()
+          title: 'Dry Run'
+          onHide: cleanup
 
   @invokeDryRun: (url, data, callback) ->
     $('body').css(cursor: 'progress')
-    $.ajax type: 'POST', url: url, dataType: 'json', data: data
```

Comment:
To make it explicit for the reader of the code? For jQuery the `Content-Type` we are sending in the controller should be explicit enough.